### PR TITLE
Refactor VLM modules / Add InternVit submodule support

### PR DESCRIPTION
**Pull request by yaoyu-33**
# What does this PR do ?

Refactor VLM modules / Add InternVit submodule support

**Collection**: [multimodal / vlm]

# Changelog 
- Add specific line by line info of high level changes in this PR.

# Usage
* You can potentially add a usage example below

```python
# Add a code snippet demonstrating how to use this 
```

# GitHub Actions CI

The Jenkins CI system has been replaced by GitHub Actions self-hosted runners.

The GitHub Actions CI will run automatically when the "Run CICD" label is added to the PR.
To re-run CI remove and add the label again.
To run CI on an untrusted fork, a NeMo user with write access must first click "Approve and run".

# Before your PR is "Ready for review"
**Pre checks**:
- [ ] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:
- [ ] New Feature
- [ ] Bugfix
- [ ] Documentation

If you haven't finished some of the above items you can still open "Draft" PR.

## Who can review?

Anyone in the community is free to review the PR once the checks have passed. 
[Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md) contains specific people who can review PRs to various areas.

# Additional Information
* Related to # (issue)

### CODE REVIEW

**Review by Victor49152 — APPROVED** [review]

File: nemo/collections/vlm/neva/model/base.py [hunk 1]
```diff
@@ -482,16 +327,9 @@ def __init__(
...
-                sharded_state_dict = dict(state_dict=self.language_model.sharded_state_dict(prefix="module."))
-                loaded_state_dict = dist_checkpointing.load(
-                    sharded_state_dict=sharded_state_dict,
-                    checkpoint_dir=ckpt_to_weights_subdir(config.language_model_from_pretrained, is_saving=False),
-                    validate_access_integrity=False,
-                )
-                loaded_state_dict = {k.removeprefix("module."): v for k, v in loaded_state_dict["state_dict"].items()}
-                self.language_model.load_state_dict(loaded_state_dict)
-                logging.info(f"Restored language model weights from {config.language_model_from_pretrained}")
+            restore_model_weights(self.language_model, config.language_model_from_pretrained)
```
Comment (yashaswikarnati) [inline nemo/collections/vlm/neva/model/base.py:330]: wonder if we should add restore model weights for vision model as well

**Review by yashaswikarnati — APPROVED** [review]

**Review by yashaswikarnati — APPROVED** [review]

**Review by suiyoubi — APPROVED** [review]

**Review by suiyoubi — APPROVED** [review]
